Assets/MixedRealityToolkit/MSBuild/Scripts/MSBuildTools.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if UNITY_EDITOR
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Compilation;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.MSBuild
+{
+    /// <summary>
+    /// Class that exposes the MSBuild project generation operation.
+    /// </summary>
+    public static class MSBuildTools
+    {
+        private static readonly HashSet<BuildTarget> supportedBuildTargets = new HashSet<BuildTarget>()
+        {
+            BuildTarget.StandaloneWindows,
+            BuildTarget.StandaloneWindows64,
+            BuildTarget.iOS,
+            BuildTarget.Android,
+            BuildTarget.WSAPlayer
+        };
+
+        public const string CSharpVersion = "7.3";
+        public const string DefaultMinUWPSDK = "10.0.14393.0";
+
+        private static readonly string uwpMinPlatformVersion = EditorUserBuildSettings.wsaMinUWPSDK;// "10.0.14393.0";
+        private static readonly string uwpTargetPlatformVersion = EditorUserBuildSettings.wsaUWPSDK;
+
+        [MenuItem("MSBuild/Generate C# SDK Projects")]
+        public static void GenerateSDKProjects()
+        {
+            // Create a copy of the packages as they might change after we create the MSBuild project
+            MakePackagesCopy();
+
+            Utilities.EnsureCleanDirectory(Application.dataPath.Replace("Assets", "MRTKBuild"));
+
+            string path = Application.dataPath.Replace("Assets", "MSBuild");
+            try
+            {
+                Utilities.EnsureCleanDirectory(path);
+            }
+            catch (IOException ex)
+            {
+                if (ex.Message.Contains(@".vs\MRTK\v15\Server\sqlite3\db.lock"))
+                {
+                    Debug.LogError("Generated project appears to be still open with Visual Studio.");
+                    throw new InvalidDataException("Generated project appears to be still open with Visual Studio.", ex);
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            List<CompilationPlatformInfo> platforms = CompilationPipeline.GetAssemblyDefinitionPlatforms()
+                .Where(t => supportedBuildTargets.Contains(t.BuildTarget))
+                .Select(CompilationPlatformInfo.GetCompilationPlatform)
+                .ToList();
+
+            CompilationPlatformInfo editorPlatform = CompilationPlatformInfo.GetEditorPlatform();
+
+            string propsOutputFolder = Application.dataPath.Replace("Assets", "MSBuild");
```

Comment:
>Application.dataPath.Replace("Assets", "MSBuild"); [](start = 39, length = 50)

This should be the same as "path" above right? I don't think path gets edited or anything